phlex/core/filter.cpp

```diff
@@ -81,8 +81,9 @@ namespace phlex::experimental {
       for (std::size_t i = 0ull; i != nargs_; ++i) {
         downstream_ports_[i]->try_put({stores[i], eom, msg_id});
       }
+      // Decision must be erased while access is claimed
```

Comment:
The bug fix itself seems good. Holding a lock while you erase the member is correct.

The other lock might be improved. I am not sure what exactly is being protected by the ```data_map::accessor a;``` but I think you might add a ```{``` before the line defining it and closing it ```}``` just before this line. Keeping that lock for less time might be a small improvement.